src/Mvc/Mvc.Core/src/ControllerBase.cs

```diff
@@ -189,6 +191,28 @@ public IObjectModelValidator ObjectValidator
             }
         }
 
+        public ProblemDetailsFactory ProblemDetailsFactory
+        {
+            get
+            {
+                if (_problemDetailsFactory == null)
+                {
+                    _problemDetailsFactory = HttpContext?.RequestServices?.GetRequiredService<ProblemDetailsFactory>();
```

Comment:
This could end up being `null` which would lead to a NRE below.

Reply:
It's expected to only ever be null in test scenarios. Basically following the same pattern as https://github.com/aspnet/AspNetCore/pull/12298/files#diff-8aa2822b0a420b6f8a04153ca4312519R172

Reply:
Wouldn't it be better to check if it's null and throw a better exception then? 🤔

Reply:
@khellang I'd rather keep the code comparable to the other properties for now. I did venture at using nullable types at one point and that forces some cleaning up some of these.

Reply:
I'm not saying you should change this property, I'm saying you might want to guard against `null` at the call site. Haven't checked usage of the other properties, but I'd consider those potential bugs as well if not properly guarded against `null`.